### ISSUE TITLE
RocketTheme's File class incorrectly assesses writable state

### ISSUE THREAD

**Issue by jgonyea** [getgrav/grav#2388]
The way the return statement will erroneously return `true` if the folder is writable but the file in question is not.

```
    /**
     * Check if file can be written.
     *
     * @return bool
     */
    public function writable()
    {
        return is_writable($this->filename) || $this->writableDir(dirname($this->filename));
    }
```

Should the boolean be an `&&` vs `||` ?

**Comment by jgonyea** [comment]
Found in vendor\rockettheme\toolbox\File\src\File.php:240

**Comment by rhukster** [comment]
As this is in Toolbox, can you take a look @mahagr ?

**Comment by mahagr** [comment]
It works as intended, if you can write to the folder, you can also write to the file. In fact I think it looks up towards the root directory if you can create the file.

**Comment by jgonyea** [comment]
@mahagr
The current logic is fine if we're talking about about a new file with a writable folder, but if an existing file somehow is read-only (with the folder being writable), then the current logic will incorrectly return true

**Comment by mahagr** [comment]
Oh, right, fixing. :)

**Comment by mahagr** [comment]
THe fix has been released as 1.6.0-rc.4
